### PR TITLE
Improve configuration process

### PR DESCRIPTION
**Pull request by ivanzy**
This PR implements 4 features mentioned in the issue #25 . Namely:

- [x] Refactoring code to have a structured config model
- [x] Support different development environments: the `NODE_ENV` variable defines which .env file should be loaded from the envs directory.
- [x] Hooks on package.json: the prefix `NODE_ENV=environment` was added in the run:dev and run:prod
- [x] Review and check the defaults: A class was created that holds the default values of al configs utilized in Zion.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -21,5 +21,6 @@ module.exports = {
     '@typescript-eslint/explicit-module-boundary-types': 'off',
     '@typescript-eslint/interface-name-prefix': 'off',
     '@typescript-eslint/no-explicit-any': 'off',
+    '@typescript-eslint/no-namespace': 'warn'
```
Comment (hyperloris) [inline .eslintrc.js:24]: I would remove this rule since we no longer use namespaces to organize defaults.

Comment (hyperloris) [inline .eslintrc.js:24]: I would remove this rule since we no longer use namespaces to organize defaults.

**Review by hyperloris — CHANGES_REQUESTED** [review]
Thank you for your contribution!
There are two minor issues that need to be addressed before merging the PR:
1. In `src/config/interfaces` you are using a lowerCamelCase convention for file names. In this repository we are adopting the same convention used by NestJS. You can read more in the [CONTRIBUTING](https://github.com/vaimee/zion/blob/main/CONTRIBUTING.md#file-names) document.
2. The env files are inside `src/config/env`. I suggest moving them to the root of the repository. To be more precise, I would move the `.env` file to the root level and `test.env` inside the `test` folder so that the "testing block" is treated as a separate application.

**Comment by ivanzy** [comment]
I do not agree on the second point. The idea is that the env variables should not be scatter over different directories. I know that if we have only a template .env and a test.env this would not be a problem, but we envision multiple possible enviroments, such as production, staging and more. Then, Where should we place the .env?

The use of a envs directory is not so rare in NodeJS systems, they do it in those tutorials: https://blog.devgenius.io/environment-variables-in-nest-js-b989bb0370bf and https://dev.to/pitops/managing-multiple-environments-in-nestjs-71l

**Comment by relu91** [comment]
> I do not agree on the second point. The idea is that the env variables should not be scatter over different directories. I know that if we have only a template .env and a test.env this would not be a problem, but we envision multiple possible enviroments, such as production, staging and more. Then, Where should we place the .env?

I'm on call right now, and I might extend this answer later but I would simply say: in the root folder. The test.env is just a special case for our integration test. It is not a real testing environment (like it might be a server where we do a couple of benchmarks or other validations). If that still bothers you I would propose the following:
- both env files in the root folder
- both env files inside a `/config`

**Review by relu91 — CHANGES_REQUESTED** [review]
Can you rename config interfaces from <name>.ts to <name>.config.ts, I think config word in this case specifies the scope of the file content.

**Comment by ivanzy** [comment]
I addressed both the requested changes in the last commits: 
-  config interfaces were renamed to `*.config.ts` 
-  .envs files are in the root

**Review by hyperloris — APPROVED** [review]

**Pull request merged**